### ISSUE TITLE
Hot exit option for MacOS window "close"

### ISSUE THREAD

**Issue by digitaltos** [microsoft/vscode#227402]
Use case: I mainly use VS Code as text editor on files that I never save as they usually only necessary for a few days. Typically using a single window too. I write notes, dump huge JSON data in it for quick analysis, compare text and use various formatting plugins, etc. 

The flow I am currently unable to achieve in MacOS is the following: 
1. I work on some files.
2. I hit the red "close" button on the top left of the window
3. VS Code does *not* prompt for saving the files
4. Once I reopen VS Code, my data is immediately there, all my previous files are open.

Step number 3 would be the hot exit option, but it does not work if I am not inside a folder. 
Step number 4 is not possible to my knowledge. I always get a fresh, empty window if I closed the window with the red button on the top left. If I was working in a folder previously, I could open that folder again, but that's a lot of manual work if I want to quickly check something. 

To my understanding, a user can do 3 actions to a given MacOS window: 

- Minimize it (yellow button on the top left), which works just as I expect. No prompt, session restored on startup. 
- Close it (red button on the top left), which closes the **window** but not the **application**. VS Code is still running at this state. This is where, no matter the hot exit settings, I will get a "save" prompt when closing the window. I don't want to save. I'm not quitting the application. 
- Quit the application (Command + Q) which also closes the window. It works just as I expected. 

To summarize, I would like to have the convenience of closing the VS Code window with the red button on the top left (as I close my other apps' windows) without needing to save all my temporary files or need to think about folder management. No prompt, session restored on startup.

**Comment by IllusionMH** [comment]
Have you tried setting `"files.hotExit": "onExitAndWindowClose"`?

**Comment by digitaltos** [comment]
> Have you tried setting `"files.hotExit": "onExitAndWindowClose"`?

Yes. To elaborate, all the observed behavior I have mentioned was with `"files.hotExit": "onExitAndWindowClose"`. 

Note that the tooltip specifically says that WindowClose is only relevant if files are open **in a folder**. I assume a conscious decision was made to not support hot exit for non-foldered files, but I could not find an issue ticket in Github regarding this. This is why I have decided to raise this feature request, to re-evaluate the need for this (and to see how relevant this use case is).

**Comment by kakulukia** [comment]
I am just witnessing the same problem here. @digitaltos did you find a solution for that?
Oh, and i also can confirm that onExitAndWindowClose DOES actually work if code is opened in a folder not just with random files. Did not notice that myself as i rarely have unsaved files when working in a folder. BUT as you have mentioned I also work that way of having random stuff open, that i really dont want to save somewhere but also dont wanna discard just yet. 

A pity, that its not working as expected. 
I will stay subscribed to that issue and will be available for voting.

Also found a lot of related issues - maybe some more voters? Maybe someone has figured out a way?

https://github.com/microsoft/vscode/issues/227402
https://github.com/microsoft/vscode/issues/173462
https://github.com/microsoft/vscode/issues/164738
https://github.com/microsoft/vscode/issues/150810
https://github.com/microsoft/vscode/issues/142021
https://github.com/microsoft/vscode/issues/133125
https://github.com/microsoft/vscode/issues/108693
https://github.com/microsoft/vscode/issues/13733
https://github.com/microsoft/vscode/issues/207
https://github.com/microsoft/vscode/issues/207

There even was a PR claiming to do what is asked for: https://github.com/microsoft/vscode/pull/16303 but i could not really follow the conversion and why its not working - BUT it might be that this broke some time after implementing it 9 years ago?

@bpasero do you remember?

**Comment by kakulukia** [comment]
How do you understand the linked PR and its conversation - should it solve our problem? Did it brake during the last 9 years after merging it?
